dstep/translator/Record.d

```diff
@@ -59,9 +61,17 @@ private:
 
     string writeRecord (string name, void delegate (Data context) dg)
     {
+        auto known = cursor.definition in this.record_definitions;
+        if (known)
+            return "";
+
         auto context = new Data;
-        static if (is(typeof(context.isFwdDeclaration) == bool))
-            context.isFwdDeclaration = !cursor.isDefinition;
+        
+        if (cursor.isDefinition)
+            this.record_definitions[cursor] = true;
+        else
+            context.isFwdDeclaration = true;
+
```

Comment:
The lines I've linked above are the lines responsible for translating the actual type declaration. The [lines below](https://github.com/jacob-carlborg/dstep/blob/master/dstep/translator/Record.d#L44-L45) are for the field declaration.

This is the flow of for translating a struct:

0. All top level nodes are iterated and translated, [here](https://github.com/jacob-carlborg/dstep/blob/master/dstep/translator/Translator.d#L62-L68)
1. When a struct is seen, it will start translating that, [here](https://github.com/jacob-carlborg/dstep/blob/master/dstep/translator/Translator.d#L140)
3. `writeRecord` is called which will translate the identifier of the struct, [here](https://github.com/jacob-carlborg/dstep/blob/master/dstep/translator/Record.d#L60)
4. `writeRecord` then calls the delegate which will start iterating all the children nodes of the struct, [here](https://github.com/jacob-carlborg/dstep/blob/master/dstep/translator/Record.d#L31)
5. When a field is seen it will check if the type is not exposed and has a declaration, [here](https://github.com/jacob-carlborg/dstep/blob/master/dstep/translator/Record.d#L37)
6. If the type is not exposed and has a declaration it will translate the declaration of the type [here](https://github.com/jacob-carlborg/dstep/blob/master/dstep/translator/Record.d#L41)
7. If the declaration is a struct it will start from the beginning at step 1